R/get_webtext.R

```diff
@@ -0,0 +1,357 @@
+  target_end_date_2wk_ahead <- format(
+    ensemble_us_1wk_ahead$target_end_date + lubridate::weeks(1),
+    "%B %d, %Y"
+  )
+
+  last_reported_target_data <- target_data |>
+    dplyr::filter(
+      week_ending_date == max(week_ending_date),
+      location == "US"
+    ) |>
+    dplyr::mutate(
+      week_end_date_formatted = format(week_ending_date, "%B %d, %Y")
+    )
+
+  last_reported_admissions <- round(last_reported_target_data$value, -2)
+
+  web_text <- glue::glue(
+    "The {hub_name} ensemble's one-week-ahead forecast predicts that the number ",
+    "of new weekly laboratory-confirmed {disease_name} hospital admissions will be ",
+    "approximately {median_forecast_1wk_ahead} nationally, with ",
+    "{lower_95ci_forecast_1wk_ahead} to {upper_95ci_forecast_1wk_ahead} ",
+    "laboratory confirmed {disease_name} hospital admissions likely reported in the ",
+    "week ending {target_end_date_1wk_ahead}. This is compared to the ",
+    "{last_reported_admissions} admissions reported for the week ",
+    "ending {last_reported_target_data$week_end_date_formatted}, the most ",
+    "recent week of reporting from U.S. hospitals.\n\n",
+    "Reported and forecasted new {disease_name} hospital admissions as of ",
+    "{forecast_due_date}. This week, {weekly_num_teams} modeling groups ",
+    "contributed {weekly_num_models} forecasts that were eligible for inclusion ",
+    "in the ensemble forecasts for at least one jurisdiction.\n\n",
+    "The figure shows the number of new laboratory-confirmed {disease_name} hospital ",
+    "admissions reported in the United States each week from ",
+    "{first_target_data_date} through {last_target_data_date} and forecasted ",
+    "new {disease_name} hospital admissions per week for this week and the next ",
+    "2 weeks through {target_end_date_2wk_ahead}.\n\n",
+    "{reporting_rate_flag}\n",
+    "Contributing teams and models:\n\n",
+    "Models included in the {hub_name} ensemble:\n",
+    "{paste(model_incl_in_hub_ensemble, collapse = '\n')}\n\n",
+    "Models not included in the {hub_name} ensemble:\n",
+    "{paste(model_not_incl_in_hub_ensemble, collapse = '\n')}"
+  )
+
+  return(web_text)
+}
+
+
+#' Generate and save text content for forecast hub
+#' visualization webpage.
+#'
+#' Light wrapper function that generates formatted text
+#' summaries and saves them to disk.
+#'
+#' @param reference_date Character, the reference date for
+#' the forecast in YYYY-MM-DD format (ISO-8601).
+#' @param disease Character, disease name ("covid" or
+#' "rsv"). Used to derive hub name, file prefix, and
+#' disease display name.
+#' @param base_hub_path Character, path to the forecast hub
+#' directory.
+#' @param hub_reports_path Character, path to forecast hub
+#' reports directory.
+#' @param excluded_locations Character vector of location
+#' codes to exclude from reporting calculations. Default:
+#' character(0).
+#'
+#' @export
+get_webtext <- function(
```

Comment:
Let's use `write_` consistently for functions that write files to disk

```suggestion
write_webtext <- function(
```